brave/src/main/java/brave/Tracer.java

```diff
@@ -495,6 +492,46 @@ public ScopedSpan startScopedSpan(String name) {
     return startScopedSpanWithParent(name, null);
   }
 
+  /**
+   * Like {@link #startScopedSpan(String)} except when there is no trace in process, the sampler
+   * {@link SamplerFunction#trySample(Object) triggers} against the supplied argument.
+   *
+   * @param name the {@link Span#name(String) span name}
+   * @param samplerFunction invoked if there's no {@link CurrentTraceContext#get() current trace}
+   * @param arg parameter to {@link SamplerFunction#trySample(Object)}
+   * @see #nextSpan(SamplerFunction, Object)
+   * @since 5.8
+   */
+  public <T> ScopedSpan startScopedSpan(String name, SamplerFunction<T> samplerFunction, T arg) {
+    if (name == null) throw new NullPointerException("name == null");
+    return newScopedSpan(name, nextContext(samplerFunction, arg));
+  }
+
+  /**
+   * Like {@link #nextSpan()} except when there is no trace in process, the sampler {@link
+   * SamplerFunction#trySample(Object) triggers} against the supplied argument.
+   *
+   * @param samplerFunction invoked if there's no {@link CurrentTraceContext#get() current trace}
+   * @param arg parameter to {@link SamplerFunction#trySample(Object)}
+   * @see #startScopedSpan(String, SamplerFunction, Object)
+   * @see #nextSpan(TraceContextOrSamplingFlags)
+   * @since 5.8
+   */
+  public <T> Span nextSpan(SamplerFunction<T> samplerFunction, T arg) {
```

Comment:
This method is used directly by the http client implementation now. It could easily be used by RPC also except we have no RPC samplers yet :P